frontend/src/components/Discover_Event/filter-section/FilterSection.tsx

```diff
@@ -20,21 +49,46 @@ const FilterSection = () => {
       ) {
         setShowCalendar(false);
       }
+      if (
+        catergoryRef.current &&
+        !(catergoryRef.current as HTMLElement).contains(event.target as Node)
+      ) {
+        setshowCatergory(false);
+      }
     }
 
     document.addEventListener("mousedown", handleClickOutside);
     return () => {
       document.removeEventListener("mousedown", handleClickOutside);
     };
-  }, [calendarRef]);
+  }, [calendarRef, catergoryRef]);
 
   return (
     <div className="filter-section-container relative">
-      <div className=" filter-section fixed bottom-7">
+      <div className=" filter-section top-[152px] bottom-7">
         <p>Filter</p>
-        <div className="filter-category">
-          <h3>Category</h3>
-          <div>
+        <div className="filter-catergory">
+          <Button
+            onClick={catergoryShow}
+            className="bg-transparent border-none p-0 shadow-none hover:bg-transparent mx-0 text-[12pt] text-black"
+          >
+            Catergory
+          </Button>
+          {showCatergory && (
+            <ScrollArea ref={catergoryRef} className="h-72 rounded-md border">
+              {categories.map((cat, index) => (
```

Comment:
Missing key props

![Screenshot 2024-05-09 at 1 01 13 AM](https://github.com/Thangnguyenvn98/GatheringGlobe/assets/69278249/5a326d20-f84f-4e10-8777-975668d1c1b4)

Reply:
Key props added for radio buttons
<img width="788" alt="image" src="https://github.com/Thangnguyenvn98/GatheringGlobe/assets/129610318/f1016263-c0ba-4ff4-a1d1-adf6b8ab5ad4">
<img width="785" alt="image" src="https://github.com/Thangnguyenvn98/GatheringGlobe/assets/129610318/09fae8e2-7b20-47fd-b35d-25c677f40e01">